### PR TITLE
Do not run CI on forked repos

### PR DESCRIPTION
**Pull request by janeyx99**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [x] other (please add here)

Please link to any issues this PR addresses.

https://github.com/janeyx99/torchtune/actions/runs/11004925000 Every night I get 2 spammy emails telling me my CI doesn't work, but like duh it doesn't work on my fork. Thanks @felipemello1  for explaining the problem, and here's the PR to gate CI behind this repo! 

FYI this experience is not very fun for OSS contributors because CI will run on their forks, and they will either get errors, or compute will be wasted on running the periodic CI. 

#### Changelog
What are the changes made in this PR?
* Mentioned above

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [x] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by felipemello1** [comment]
Thanks, @janeyx99! You are a champ! Do you know the best way to test it? Just let the CI run?

**Comment by janeyx99** [comment]
@felipemello1 Yes, nothing should be changed in torchtune CI (and I should stop getting emails).

File: .github/workflows/build_docs.yaml [hunk 1]
```diff
@@ -82,7 +83,7 @@ jobs:
   upload:
     runs-on: ubuntu-latest
     needs: build_docs
-    if: github.repository == 'pytorch/torchtune' && github.event_name == 'push' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v'))
+    if: github.repository_owner == 'pytorch' && github.event_name == 'push' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v'))
```
Comment (ebsmothers) [inline .github/workflows/build_docs.yaml:86]: I'm curious about this one in particular.. why isn't `github.repository == 'pytorch/torchtune'` sufficient? My assumption is that forks would have a separate repository name, is that not the case?

**Review by ebsmothers — APPROVED** [review]
This is why I was getting all those emails?! Seriously though.. thank you for fixing this

File: .github/workflows/build_docs.yaml (hunk 1 above)
Comment (janeyx99) [reply .github/workflows/build_docs.yaml:86]: it totally was sufficient, i just changed it for consistency so someone looking can see that it’s the same across the workflows

**Pull request merged**